palette/src/encoding/rec_standards.rs

```diff
@@ -0,0 +1,291 @@
+    type Primaries = Rec2020;
+    type WhitePoint = D65;
+
+    #[rustfmt::skip]
+    #[inline(always)]
+    fn rgb_to_xyz_matrix() -> Option<Mat3<f64>> {
+        // Matrix calculated using specified primary values and white point
+        // using formulas from http://www.brucelindbloom.com/index.html?Eqn_RGB_XYZ_Matrix.html
+        Some([
+            0.6370102, 0.1446150, 0.1688448,
+            0.2627217, 0.6779893, 0.0592890,
+            0.0000000, 0.0280723, 1.0607577,
+        ])
+    }
+
+    #[rustfmt::skip]
+    #[inline(always)]
+    fn xyz_to_rgb_matrix() -> Option<Mat3<f64>> {
+        // Matrix calculated using specified primary values and white point
+        // using formulas from http://www.brucelindbloom.com/index.html?Eqn_RGB_XYZ_Matrix.html
+        Some([
+             1.7165107, -0.3556417, -0.2533455,
+            -0.6666930,  1.6165022,  0.0157688,
+             0.0176436, -0.0427798,  0.9423051,
+        ])
+    }
+}
+
+impl RgbStandard for Rec2020 {
+    type Space = Rec2020;
+    type TransferFn = RecOetf;
+}
+
+impl LumaStandard for Rec2020 {
+    type WhitePoint = D65;
+    type TransferFn = RecOetf;
+}
+
+/// The Rec. 709 standard, color space, and transfer function.
+///
+/// # As transfer function
+///
+/// `Rec709` will not use any kind of approximation when converting from `T` to
+/// `T`. This involves calls to `powf`, which may make it too slow for certain
+/// applications.
+pub struct Rec709;
+
+impl RgbStandard for Rec709 {
+    type Space = Srgb;
+    type TransferFn = RecOetf;
+}
+
+impl LumaStandard for Rec709 {
+    type WhitePoint = D65;
+    type TransferFn = RecOetf;
+}
+
+/// The opto-electronic transfer function used in standard dynamic range (SDR)
+/// standards by the ITU-R such as [`Rec709`] and [`Rec2020`].
+pub struct RecOetf;
+
+const ALPHA: f64 = 1.09929682680944;
+const BETA: f64 = 0.018053968510807;
+
+impl<T> IntoLinear<T, T> for RecOetf
+where
+    T: Real + Powf + MulAdd + Arithmetics + PartialCmp + Clone,
+    T::Mask: LazySelect<T>,
+{
+    #[inline]
+    fn into_linear(encoded: T) -> T {
+        lazy_select! {
+            if encoded.lt(&T::from_f64(4.5*BETA)) => T::from_f64(1.0 / 4.5) * &encoded,
+            else => encoded.clone().mul_add(T::from_f64(1.0 / ALPHA), T::from_f64(1.0 - 1.0 / ALPHA)).powf(T::from_f64(1.0 / 0.45))
+        }
+    }
+}
+
+impl<T> FromLinear<T, T> for RecOetf
+where
+    T: Real + Powf + MulSub + Arithmetics + PartialCmp + Clone,
+    T::Mask: LazySelect<T>,
+{
+    #[inline]
+    fn from_linear(linear: T) -> T {
+        lazy_select! {
+            if linear.lt(&T::from_f64(BETA)) => T::from_f64(4.5) * &linear,
+            else => linear.clone().powf(T::from_f64(0.45)).mul_sub(T::from_f64(ALPHA), T::from_f64(ALPHA - 1.0))
+        }
+    }
+}
+
+impl IntoLinear<f32, u8> for RecOetf {
+    #[inline]
+    fn into_linear(encoded: u8) -> f32 {
+        REC_OETF_U8_TO_F32[encoded as usize]
+    }
+}
+
+impl FromLinear<f32, u8> for RecOetf {
```

Comment:
I'm a bit on the fence with these ones, since they pretty much do the same thing as the generic implementations. The original intent was to only specialize when there's a significant improvement to be had. That's part of why there are no alternatives for other unsigned integers.

At the very least, I think they can delegate to the generic versions, but I'm leaning towards not including them at all, for now. :thinking:

Reply:
That's fair. I'm pretty certain I can replicate the methodology used by [`fast-srgb8`](https://crates.io/crates/fast-srgb8) for this if you think that is a good idea. Would it be preferable to do that or to simply remove these implementations for `IntoLinear` and `FromLinear` for the time being?